Sources/AblyAssetTrackingInternal/WorkerQueue/LegacyWorker.swift

```diff
@@ -0,0 +1,33 @@
+import Foundation
+
+/// A worker that runs tasks that the Publisher and Subscriber used to run asynchronously on their
+/// own dispatch queues. This worker allows us to tie up legacy work and new asynchronous
+/// work, for simplicity and consistency during the transition.
+open class LegacyWorker<PropertiesType, WorkerSpecificationType> : Worker
+{
+    let work: () -> Void
+
+    public init (work: @escaping () -> Void) {
+        self.work = work
+    }
+
+    public func doWork(properties: PropertiesType, doAsyncWork: (@escaping () throws -> Void) -> Void, postWork: @escaping (WorkerSpecificationType) -> Void) throws -> PropertiesType {
+        doAsyncWork({ [work] in
```

Comment:
I'm not sure I understand why it's `doAsyncWork` here. If the `Default{Publisher, Subscriber}`s’ `workingQueue` is being replaced by 

If we want to replace the following code from `DefaultPublisher`:

```swift
    private func performOnWorkingQueue(_ operation: @escaping () -> Void) {
        workingQueue.async(execute: operation)
    }
```

(which enqueues a piece of work to be asynchronously performed on a serial queue), then shouldn't the worker execute the `work` directly, given that `WorkerQueue`’s `enqueue` method already enqueues the worker to be performed asynchronously on its `workingQueue`?

Reply:
Ah, I went for an extra level of removal here - as the work the publisher is doing is sometimes (not always) blocking, I made it so that it would always defer to the async serial queue, so that we don't block other works on the main worker queue :) Happy to talk this through though!

Reply:
I think we came to the decision in our call today that we'd execute the work directly on the worker, right?

Reply:
Yes - changed in [71392bc](https://github.com/ably/ably-asset-tracking-swift/pull/573/commits/71392bc4a64c168a1ce148fd0837831388441168)